src/main/java/com/atlassian/jira/cloud/jenkins/listeners/ConfigurationSaveableListener.java

```diff
@@ -0,0 +1,44 @@
+package com.atlassian.jira.cloud.jenkins.listeners;
+
+import com.atlassian.jira.cloud.jenkins.pluginConfigApi.PluginConfigApi;
+import com.atlassian.jira.cloud.jenkins.util.IpAddressProvider;
+import hudson.Extension;
+import hudson.XmlFile;
+import hudson.model.Saveable;
+import hudson.model.listeners.SaveableListener;
+import javax.inject.Inject;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Extension
+public class ConfigurationSaveableListener extends SaveableListener {
+
+    private static final Logger LOGGER =
+            Logger.getLogger(ConfigurationSaveableListener.class.getName());
+
+    private transient PluginConfigApi pluginConfigApi;
+
+    @Inject
+    public void setConnectionDataApi(final PluginConfigApi pluginConfigApi) {
+        this.pluginConfigApi = pluginConfigApi;
+    }
+
+    @Override
+    public void onChange(final Saveable saveable, final XmlFile file) {
+        if (!(saveable instanceof com.atlassian.jira.cloud.jenkins.config.JiraCloudPluginConfig)) {
+            return;
+        }
+        sendPluginConfigData();
+    }
+
+    public void sendPluginConfigData() {
+        try {
+            String ipAddress = IpAddressProvider.getIpAddress();
+            LOGGER.log(Level.INFO, "IP Address: {0}", ipAddress);
+
+            // TODO - get webhhokurl/secret from save xml, fire jenkins plugin data event
+        } catch (Exception e) {
+            LOGGER.log(Level.SEVERE, "Error processing XML content", e);
```

Comment:
Again, this will only show up in logs in pipelines in our customers servers. We could leave this as is so it's always there if there's an issue. Alternatively, we could change to level to debug? That way it will only come through if the customer has checked the allow debug logging box? We probably don't want to add extra noise to customers pipeline logs and if someone calls out they can't see their IP address in J4J, we can ask them to allow debug logging temporarily?

Reply:
ahhhh thats lame, debug sounds like a good call tho. will update

Reply:
dropped the IP so it doesnt log, and happy to leave as option 1 for now.